I can confirm failure on macos 10.13:

```
tests.test_visibility.FilterEventsForServerTestCase.test_large_room
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/private/tmp/nix-build-matrix-synapse-1.4.1.drv-0/matrix-synapse-1.4.1/tests/rest/media/v1/test_url_preview.py", line 247, in test_non_ascii_preview_content_type
    self.assertEqual(channel.code, 200)
  File "/nix/store/k66dzav0xyij9wjw01f2996dyhrp5kkn-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/trial/_synctest.py", line 432, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "/nix/store/gaascisqfmxckzhvs5q1wbx6ycsbk4fi-python3-3.7.4/lib/python3.7/unittest/case.py", line 852, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/nix/store/gaascisqfmxckzhvs5q1wbx6ycsbk4fi-python3-3.7.4/lib/python3.7/unittest/case.py", line 845, in _baseAssertEqual
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: 500 != 200

tests.rest.media.v1.test_url_preview.URLPreviewTests.test_non_ascii_preview_content_type
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/private/tmp/nix-build-matrix-synapse-1.4.1.drv-0/matrix-synapse-1.4.1/tests/rest/media/v1/test_url_preview.py", line 214, in test_non_ascii_preview_httpequiv
    self.assertEqual(channel.code, 200)
  File "/nix/store/k66dzav0xyij9wjw01f2996dyhrp5kkn-python3.7-Twisted-18.9.0/lib/python3.7/site-packages/twisted/trial/_synctest.py", line 432, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "/nix/store/gaascisqfmxckzhvs5q1wbx6ycsbk4fi-python3-3.7.4/lib/python3.7/unittest/case.py", line 852, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/nix/store/gaascisqfmxckzhvs5q1wbx6ycsbk4fi-python3-3.7.4/lib/python3.7/unittest/case.py", line 845, in _baseAssertEqual
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: 500 != 200

tests.rest.media.v1.test_url_preview.URLPreviewTests.test_non_ascii_preview_httpequiv
-------------------------------------------------------------------------------
```

--- a/pkgs/servers/matrix-synapse/default.nix
+++ b/pkgs/servers/matrix-synapse/default.nix
@@ -23,11 +23,11 @@ let
 
 in buildPythonApplication rec {
   pname = "matrix-synapse";
-  version = "1.4.0";
+  version = "1.4.1";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1y8yhzsf2lk2d7v4l61rpy4918c0qz276j79q88l9yazb6gw5pkk";
+    sha256 = "10vfjf5f4micqwpxnw2dliq4y0j0mqj0087p6kgnfi5w9syz8g75";
   };
 
   patches = [
@@ -74,6 +74,8 @@ in buildPythonApplication rec {
   ] ++ lib.optional enableSystemd systemd;
 
   checkInputs = [ mock parameterized openssl ];
+
+  doCheck = !stdenv.isDarwin;
 
   checkPhase = ''
     PYTHONPATH=".:$PYTHONPATH" ${python3.interpreter} -m twisted.trial tests

